src/NBitcoin/ChainBase.cs

```diff
@@ -52,41 +75,54 @@ public ChainedBlock SetTip(ChainBase otherChain)
             return this.SetTip(otherChain.Tip);
         }
 
+        /// <summary>
+        /// Sets the tip of this chain based upon another block header.
+        /// </summary>
+        /// <param name="header">The block header to set to tip.</param>
+        /// <returns>Whether the tip was set successfully.</returns>
         public bool SetTip(BlockHeader header)
         {
             ChainedBlock chainedHeader;
             return this.TrySetTip(header, out chainedHeader);
         }
 
+        /// <summary>
+        /// Attempts to set the tip of this chain based upon another block header.
+        /// </summary>
+        /// <param name="header">The block header to set to tip.</param>
+        /// <param name="chainedHeader">The newly chained block header for the tip.</param>
+        /// <returns>Whether the tip was set successfully. Will return <c>false</c> if the block header previous block link cannot be found in the current chain.</returns>
```

Comment:
hard to read the second sentence :-)
how about
`<returns>Whether the tip was set successfully. The method fails (and returns <c>false</c>) if the <paramref name="header"/>'s link to a previous header does not point to any block in the current chain.</returns>`